agent_id/CMakeLists.txt

```diff
@@ -51,12 +51,9 @@ target_link_libraries(${PROJECT_NAME}
 
 target_include_directories(${PROJECT_NAME} PUBLIC include ${catkin_INCLUDE_DIRS})
 
-# Tests for agent_id
-find_package(GTest REQUIRED)
-include_directories(
-  ${GTEST_INCLUDE_DIRS}
-)
-add_executable(${PROJECT_NAME}-tests
-  src/test/AgentIDTests.cpp
-)
-target_link_libraries(${PROJECT_NAME}-tests ${PROJECT_NAME} ${GTEST_LIBRARIES})
\ No newline at end of file
+if(CATKIN_ENABLE_TESTING)
```

Comment:
We are trying to get away from ROS/Catkin. Therefore we already try to avoid catkin specific stuff like this.

Reply:
Sure, but this is a test and should be treated as such, so that CI systems can pick it up.
Because of catkin and gtest weirdness it neeeds to run through the catkin_add_gtest macro, which in turn should be guarded by CATKIN_ENABLE_TESTING
I do not know of a working alternative for this as long as you are still building with catkin and expect catkin run_tests to work.

Reply:
Sure, but this is a test and should be treated as such, so that CI systems can pick it up.
Because of catkin and gtest weirdness it neeeds to run through the catkin_add_gtest macro, which in turn should be guarded by CATKIN_ENABLE_TESTING
I do not know of a working alternative for this as long as you are still building with catkin and expect catkin run_tests to work.